### DISCUSSION TITLE
Ampersand character being interpreted in comments in the deploy scripts

### DISCUSSION THREAD

**Discussion by ClementC** [theory/sqitch#752] [General]
Hi there, beginner users of sqitch here 👋 

We had an issue, with a comment containing a URL to some wiki page, and the URL included the ampersand character.
This caused the execution of the `sqitch deploy` command to fail, and it was not immediately clear for us what caused the failure.

My question is: is there someplace in the documentation indicating explicitly what is interpreted by sqitch at different stages, and what are the best practices when it comes to documenting the deploy scripts?
I found [this section](https://sqitch.org/docs/manual/sqitch-add/#templates) to be somewhat related, but I felt it was not really answering the question I had.

**Comment by theory** [comment]
That's odd. Can you paste an example of the problematic line? 

The Plan parser defines the format of a plan, but a note is allowed to be any character at all aside from a newline, which can be stored as the literal strings `\r` or `\r\n`. The regular expression that matches notes is [here](/sqitchers/sqitch/blob/afe87ce/lib/App/Sqitch/Plan.pm#L204).

**Comment by ClementC** [comment]
Sure, here's a sanitized version of the deploy script:

```sql
-- Deploy misc script
-- Documentation link: https://jira_url.com/wiki/pages/viewpage.action?spaceKey=SPACE_NAME&title=PageTitle

USE WAREHOUSE &warehouse;

CREATE TABLE IF NOT EXISTS A.B (
    C DATE,
    D NUMBER(38,0)
);
```

This was causing an error with the following message:
`Variable title is not defined`

**Comment by theory** [comment]
Oh, I thought you mean in the plan file. This syntax is defined by the SnowSQL client. Note the `&warehouse` variable (which presumably you're setting on deploy). To use variables, you'll need to not use `&` in your snowSQL scripts. From [the docs](https://docs.snowflake.com/en/user-guide/snowsql-use#substituting-variables-in-a-session), it looks like you can escape the variable with… an ampersand, e.g., `&&title=`.

**Comment by ClementC** [comment]
Ok, thanks a lot then, you gave me all the info I needed! 👍 
I'm closing the issue.

**Comment by theory** [comment]
You can also edit that line in the file directly, It's just informational, doesn't affect any business logic at all aside from the hash digest used to identify the file itself (so don't change it after you've tagged and released).

Mind if I move this issue to discussions? Might help others in the future.

**Comment by ClementC** [comment]
Yup, on our end it just means that having URLs in the scripts are a bad habit, and in the future it's better to put them in the plan for instance, where they would not be parsed in the comments.

No problem to move it to Discussions! And again thanks a lot for your quick and efficient answers! 🙂

**Reply by theory** [reply]
I think URLs in scripts are fine, you just need to escape ampersands. Not sure wha you mean by "in the plan", because the note for a change in the plan is used for a comment in the scripts, although you can change the template to omit them. Brief intro to change file templates [here](https://justatheory.com/2013/09/sqitch-templating/).

**Reply by ClementC** [reply]
I meant in the `sqitch.plan` file.
Good to know that the note is used in the script as well.
Thanks a lot! 🙂
